### PR TITLE
refactor: write kzip always for JavaStandalone extractor

### PR DESCRIPTION
**Pull request by danielmoy-google**
Follow-up to #3297, getting rid of kindex and indexpack support.

Relevantly, this will mean that setting KYTHE_OUTPUT_DIRECTORY now
writes potentially multiple kzips, instead of kindex.  We can bundle
those together later with zipmerge if we want.

### CODE REVIEW

File: kythe/java/com/google/devtools/kythe/extractors/java/standalone/AbstractJavacWrapper.java [hunk 1]
```diff
@@ -122,18 +121,25 @@ private static void outputIndexInfo(CompilationDescription indexInfo) throws IOE
       if (outputFile.endsWith(IndexInfoUtils.KZIP_FILE_EXT)) {
         IndexInfoUtils.writeKzipToFile(indexInfo, outputFile);
       } else {
-        IndexInfoUtils.writeKindexToFile(indexInfo, outputFile);
+        System.err.printf("Unsupported output file: %s%n", outputFile);
```
Comment (salguarnieri) [inline kythe/java/com/google/devtools/kythe/extractors/java/standalone/AbstractJavacWrapper.java]: %n -> \n ?

File: kythe/java/com/google/devtools/kythe/extractors/java/standalone/AbstractJavacWrapper.java [hunk 2]
```diff
@@ -122,18 +121,25 @@ private static void outputIndexInfo(CompilationDescription indexInfo) throws IOE
...
-    new Archive(outputDir).writeDescription(indexInfo);
+    // Just rely on the underlying compilation unit's signature to get the filename, if we're not
+    // writing to a single kzip file.
+    String name =
+        indexInfo
+            .getCompilationUnit()
+            .getVName()
+            .getSignature()
+            .trim()
+            .replaceAll("^/+|/+$", "")
```
Comment (salguarnieri) [inline kythe/java/com/google/devtools/kythe/extractors/java/standalone/AbstractJavacWrapper.java]: Do these replaces create a risk of name collision?

**Comment by danielmoy-google** [comment]
Ah this will probably be required for #3264 if I'm remembering correctly.

File: kythe/java/com/google/devtools/kythe/extractors/java/standalone/AbstractJavacWrapper.java (hunk 1 above)
Comment (danielmoy-google) [reply kythe/java/com/google/devtools/kythe/extractors/java/standalone/AbstractJavacWrapper.java]: I'm unsure here.  The rest of this file is already using %n, so I'd like to keep it that way for now.  If we want to move it all to \n, rather do that as a separate refactor step, and all at once (and maybe other files as well?).

In any event, %n is more platform-agnostic, I think?  (Windows?)

File: kythe/java/com/google/devtools/kythe/extractors/java/standalone/AbstractJavacWrapper.java (hunk 2 above)
Comment (danielmoy-google) [reply kythe/java/com/google/devtools/kythe/extractors/java/standalone/AbstractJavacWrapper.java]: In theory maybe?  In practice, this is what we already used before.  So, might be ok.

File: kythe/java/com/google/devtools/kythe/extractors/java/standalone/AbstractJavacWrapper.java (hunk 1 above)
Comment (danielmoy-google) [reply kythe/java/com/google/devtools/kythe/extractors/java/standalone/AbstractJavacWrapper.java]: Yup ok, from the java docs, %n is the platform-agnostic one

https://docs.oracle.com/javase/tutorial/java/data/numberformat.html

Comment (salguarnieri) [reply kythe/java/com/google/devtools/kythe/extractors/java/standalone/AbstractJavacWrapper.java]: TIL

File: kythe/java/com/google/devtools/kythe/extractors/java/standalone/AbstractJavacWrapper.java (hunk 2 above)
Comment (salguarnieri) [reply kythe/java/com/google/devtools/kythe/extractors/java/standalone/AbstractJavacWrapper.java]: sgtm

**Comment by danielmoy-google** [comment]
Ok should have a fix for the failing test - kzip writing needs to populate parent directory too, just like kindex.

We just hadn't run into this problem yet, because all our kzip writing has been done from more precise wrapping logic that creates the output directory ahead of time.

**Pull request merged**